### PR TITLE
lib: modem_info: Update RSRP/RSRQ calculation

### PR DESCRIPTION
**Pull request by trantanen**
AT Command Reference Guide has been updated based on how the modem behaves. Doing corresponding update to app side, i.e., modem_info and lte_link_control.

Jira: TNSW-62531

### CODE REVIEW

File: include/modem/modem_info.h
Comment (tokangas) [inline include/modem/modem_info.h:34]: As discussed, these should not be used directly, so these can be removed and the values moved into the conversion macros.

File: include/modem/modem_info.h
Comment (tokangas) [inline include/modem/modem_info.h:77]: Please modify comment to only describe RSRQ conversion and do the same for the RSRP macro comment.

File: include/modem/modem_info.h
Comment (trantanen) [inline include/modem/modem_info.h:53]: I think I will add the formula from the AT command reference guide when I get the latest draft and the wording in there.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:738]:
```suggestion
    * The formulas of RSRP and RSRQ values in :c:macro:`RSRP_IDX_TO_DBM` and :c:macro:`RSRQ_IDX_TO_DB` based on AT command reference guide updates.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:743]:
```suggestion
    Clients should have used the :c:macro:`RSRP_IDX_TO_DBM` and the :c:macro:`RSRQ_IDX_TO_DB` macros.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:739]:
```suggestion
      The formulas are now aligned with the modem implementation that has not changed
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst
Comment (tokangas) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:764]:
```suggestion
  * Removed ``RSRP_OFFSET_VAL``, ``RSRQ_OFFSET_VAL`` and ``RSRQ_SCALE_VAL`` from the API.
```

File: include/modem/modem_info.h
Comment (tokangas) [inline include/modem/modem_info.h:53]: This should be dBm instead, but should it be plural, @peknis?
```suggestion
 * The index value of RSRP can be converted to dBm with the following formula:
```

File: include/modem/modem_info.h
Comment (tokangas) [inline include/modem/modem_info.h:56]:
```suggestion
 * * index < 0: index – 140
 * * index = 0: Not used
 * * index > 0: index – 141
```

File: include/modem/modem_info.h
Comment (tokangas) [inline include/modem/modem_info.h:81]: Should "decibel" be plural or not, @peknis?

File: include/modem/modem_info.h
Comment (tokangas) [inline include/modem/modem_info.h:85]:
```suggestion
 * * index < 0: (index – 39) / 2
 * * index = 0: Not used
 * * index > 0 and index < 35: (index – 40) / 2
 * * index ≥ 35: (index – 41) / 2
```

File: lib/modem_info/modem_info.c
Comment (tokangas) [inline lib/modem_info/modem_info.c:876]: This applies to other places where the macros are used as well, but the value returned by the macro is a float, and it's assigned into an int, so probably it should be rounded appropriately.

File: include/modem/modem_info.h
Comment (trantanen) [reply include/modem/modem_info.h:53]: This is copy paste from AT command spec and I did send the question to Jaakko earlier

File: include/modem/modem_info.h
Comment (trantanen) [reply include/modem/modem_info.h:81]: Waiting for Jaakko's feedback

File: lib/modem_info/modem_info.c
Comment (trantanen) [reply lib/modem_info/modem_info.c:876]: Maybe. As discussed this is how it's been and after all, this doesn't really affect on any functionality.

File: include/modem/modem_info.h
Comment (peknis) [reply include/modem/modem_info.h:81]: Can be either.

File: include/modem/modem_info.h
Comment (peknis) [reply include/modem/modem_info.h:53]: Can be either singular or plural. Not a big deal.

**Review by peknis — APPROVED** [review]

**Review by maxd-nordic — APPROVED** [review]

File: include/modem/modem_info.h
Comment (trantanen) [reply include/modem/modem_info.h:53]: I'll use dBm in the text because decibel from AT spec is just wrong.

File: include/modem/modem_info.h
Comment (trantanen) [reply include/modem/modem_info.h:81]: Changing this to dB to be aligned with previous macro

File: tests/lib/modem_info/src/main.c
Comment (tokangas) [inline tests/lib/modem_info/src/main.c:570]: I guess this works because -34.5 gets implicitly converted into an integer. Still, looks a bit funny. Because both macros return a float, it would make sense to test these using floats.

Comment (trantanen) [reply tests/lib/modem_info/src/main.c]: Good catch! I also added @param and @return statements into the header. Let's see from generated documentation if they look ok. If not, I'll remove the tags and include the same information in the text.

**Review by tokangas — APPROVED** [review]

**Comment by trantanen** [comment]
Please review @nrfconnect/ncs-co-networking, @nrfconnect/ncs-modem, @nrfconnect/ncs-code-owners and @bjda.

**Review by nordicjm — APPROVED** [review]

**Review by lemrey — APPROVED** [review]

**Review by eivindj-nordic — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**
